File: Level #1/Javascript-Basics-Course/JavaScript-Basics/JS-LoopsArraysStrings-Homework/_16_CardsFrequencies/cardFrequencies.js
/**
 * Created by Rosen on 18.11.2014 г..
 */
function findCardFrequency(str) {
    var cards = str.split(/[♥♣♦♠ ]+/g);
    var cardFrequencies = { };
    var totalCards = 0;

    for (var i = 0; i < cards.length; i++) {
        if(cards[i] == '') {
            continue;
        } else {
            if(cardFrequencies[cards[i]] == undefined) {
                cardFrequencies[cards[i]] = 1;
            } else {
                cardFrequencies[cards[i]]++;
            }
            totalCards++;
        }
    }

    for (var card in cardFrequencies) {
        var count = cardFrequencies[card];
        var percent = parseFloat(count)/totalCards*100;
        percent = percent.toFixed(2);
        console.log(card + ' -> ' + percent + '%');
    }
}

findCardFrequency('8♥ 2♣ 4♦ 10♦ J♥ A♠ K♦ 10♥ K♠ K♦');
console.log();
findCardFrequency('J♥ 2♣ 2♦ 2♥ 2♦ 2♠ 2♦ J♥ 2♠');
console.log();
findCardFrequency('10♣ 10♥');

File: Level #1/Java-Basics-Course/Java-Basics/JavaCollectionsBasicsHomework/CardFrequencies.java
import java.util.LinkedHashMap;
import java.util.Scanner;


public class CardFrequencies {

	public static void main(String[] args) {
		
		Scanner input = new Scanner(System.in);
		
		String[] cards = input.nextLine().split("[♥♣♦♠ ]+");
		LinkedHashMap<String, Integer> cardsFrequencies = new LinkedHashMap<>();
		int totalCards = 0;
		for (int i = 0; i < cards.length; i++) {
			if (cardsFrequencies.containsKey(cards[i])) {
				int count = cardsFrequencies.get(cards[i]);
				cardsFrequencies.put(cards[i], count + 1);
			} else {
			cardsFrequencies.put(cards[i], 1);
			}
			totalCards++;
		}
		
		for (String card : cardsFrequencies.keySet()) {
			int count = cardsFrequencies.get(card);
			double percent = (double)count / totalCards * 100;
			System.out.printf("%s -> %.2f%%%n", card, percent);
		}

	}

}
